### PR TITLE
Add squid proxy

### PR DESCRIPTION
**Pull request by zhouzhuojie**
Related to https://github.com/pytorch/pytorch/issues/61793

We are creating a simple squid cache proxy with terraform and aws resources that can help caching the common filetype of http egress traffic in pytorch CI.

### CODE REVIEW

File: aws/websites/squid-proxy/squid/squid.tf [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  image_id                     = "${var.aws_ami}"
+  instance_type                = "${var.aws_instance_type}"
+  key_name                     = "${var.aws_key_name}"
+  security_groups              = ["${var.aws_security_group_id}"]
+  associate_public_ip_address  = true
+  user_data                    = local.user_data
+  ebs_optimized = true
+  root_block_device {
+    volume_type = "gp3"
+    volume_size = 160
```
Comment (zhouzhuojie) [inline aws/websites/squid-proxy/squid/squid.tf:55]: we are using SSD EBS optimized for disk I/O performance

File: aws/websites/squid-proxy/squid/squid.tf [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  }
+}
+
+resource "aws_autoscaling_group" "asg_1" {
+  name                         = "${var.environment_name}_asg_squid_proxy"
+  launch_configuration         = "${aws_launch_configuration.lc_1.name}"
+  vpc_zone_identifier          = "${var.aws_subnet_ids}"
+  load_balancers               = ["${aws_elb.elb_1.id}"]
+  health_check_grace_period    = 180
+  health_check_type            = "ELB"
```
Comment (zhouzhuojie) [inline aws/websites/squid-proxy/squid/squid.tf:69]: With ELB and autoscaling to make sure instances are healthy - the default health check is on TCP port 3128

File: aws/websites/squid-proxy/.terraform.lock.hcl [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# This file is maintained automatically by "terraform init".
```
Comment (malfet) [inline aws/websites/squid-proxy/.terraform.lock.hcl:1]: If this file autogenerated, can you please add it to `.gitattribtues` file to that it'll be excluded from reviews

File: aws/websites/squid-proxy/.gitignore [hunk 4]
```diff
@@ -0,0 +1,6 @@
+target/
+*.iml
+.idea/
+**/.terraform/
+**/*.tfstate*
+**/.DS_Store
```
Comment (malfet) [inline aws/websites/squid-proxy/.gitignore:6]: Newline?

File: aws/websites/squid-proxy/main.tf [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  environment_name      = var.environment_name
+  aws_key_name          = var.aws_key_name
+  aws_region            = var.aws_region
+  aws_profile           = var.aws_profile
+  aws_security_group_id = aws_security_group.sg.id
+  aws_subnet_ids        = [tolist(data.aws_subnet_ids.dev.ids)[0]]
+  aws_ami               = lookup(var.aws_amis, var.aws_region)
+  aws_public_vpc_cidr   = var.aws_public_vpc_cidr
+  squid_port            = var.squid_port
+}
```
Comment (malfet) [inline aws/websites/squid-proxy/main.tf:52]: newline
```suggestion
}
```

File: aws/websites/squid-proxy/squid/outputs.tf [hunk 6]
```diff
@@ -0,0 +1,4 @@
+output "dns_name" {
+  value = "${aws_elb.elb_1.dns_name}"
+}
+
```
Comment (malfet) [inline aws/websites/squid-proxy/squid/outputs.tf:4]: Extra line
```suggestion
```

File: aws/websites/squid-proxy/squid/squid.tf [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+  namespace = "AWS/EC2"
+  period = "120"
+  statistic = "Average"
+  threshold = "80"
+  alarm_description = "Check whether EC2 instance CPU utilisation is over 80% on average"
+  alarm_actions = ["${aws_autoscaling_policy.asp_1.arn}"]
+  dimensions = {
+    AutoScalingGroupName = "${aws_autoscaling_group.asg_1.name}"
+  }
+}
```
Comment (malfet) [inline aws/websites/squid-proxy/squid/squid.tf:127]: Newline
```suggestion
}
```

**Review by malfet — APPROVED** [review]

**Comment by seemethere** [comment]
Would this be better put in https://github.com/fairinternal/pytorch-gha-infra with the other terraform? Or put into configerator?

**Pull request merged**